### PR TITLE
test(jira/MBP-29): implement api for providing dfsp hostname and ip whitelist

### PR DESCRIPTION
**Pull request by sri-miriyala**
test(jira/MBP-29): implement api for providing dfsp hostname and ip whitelist - https://modusbox.atlassian.net/browse/MBP-29
- added dockerfiles for functional tests
- updated func-test
  - env.example with missing env var
  - package.json to move test dev-dependencies to dependencies (as that is what it is actually doing)
- updated dockerignore with some more useful changes
- added new defaults and constraints to constants.js
- re-working docker run
- updated functional-test readme
  - fixed md lint issues
  - added how to run MCM API from docker
- added ci-cd badge

### CODE REVIEW

**Comment by mdebarros** [comment]
Hey @sri-miriyala ,

What are your objectives here with the functional tests?

Are you planning to add functional-tests for all the API operations?

**Comment by sri-miriyala** [comment]
Hi @mdebarros ...that is my plan..to start with add these tests for the new endpoints that you added and slowly add the functional tests for the rest of all the endpoints.

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        working-directory: ${{env.APP_WORK_DIRECTORY}}
+        run: npm i
+      - name: Install Functional-tests dependencies
+        working-directory: ${{env.TEST_FUNC_WORK_DIRECTORY}}
+        run: npm i
+
+      # Actions
+      - name: Start backends
+        working-directory: ${{env.APP_WORK_DIRECTORY}}
+        run: npm run start:backend
```
[the remark this answers is not in the document]
Comment (mdebarros) [reply .github/workflows/build-and-test.yaml:128]: No it is not.

[the remark this answers is not in the document]
Comment (mdebarros) [reply .github/workflows/build-and-test.yaml:128]: This is pretty much the same as if you started the backends on your local machine, and executed the functional tests against it.

The only difference here is that we are running a Docker image of the MCM API that the functional tests are executed against.

[the remark this answers is not in the document]
Comment (sri-miriyala) [reply .github/workflows/build-and-test.yaml:128]: Thanks. So the database(backend) is started via npm and the API is running as Docker and the tests are pointing to the Docker, right @mdebarros

[the remark this answers is not in the document]
Comment (mdebarros) [reply .github/workflows/build-and-test.yaml:128]: Yes, that is correct @sri-miriyala.

**Review by sri-miriyala — COMMENTED** [review]
LGTM

**Pull request merged**